packages/liquidator/src/liquidationStrategy.js

```diff
@@ -0,0 +1,223 @@
+const assert = require("assert");
+// This module is responsible for sizing liquidation positions.
+// If a defensive strategy is enabled, it will check withdraw positions
+// to see if they require an extension rather than full liquidation.
+module.exports = (config, libs, emit = x => x) => {
+  const {
+    // how much money to set aside for defense in tokens
+    whaleDefenseFundWei = 0,
+    // when to activate defense, when withdraw is X% in time complete
+    defenseActivationPercent,
+    liquidationMinPrice = "0",
+    // `liquidationDeadline`: Aborts the liquidation if the transaction is mined this amount of time after the
+    liquidationDeadline = 300,
+    // the minimum amount of tokens to liquidate with resetting timer
+    withdrawalLiveness
+  } = config;
+
+  const { toBN, BN } = libs;
+
+  assert(liquidationDeadline >= 0, "liquidationDeadline must be 0 or higher");
+  assert(toBN, "requires toBN");
+  assert(BN, "requires BN limit");
+
+  // these are only needed if WDF is activated
+  whaleDefenseFundWei &&
+    assert(
+      defenseActivationPercent !== null && defenseActivationPercent >= 0,
+      "Requires defenseActivationPercent to bet set between 0 and 100"
+    );
+  whaleDefenseFundWei && assert(withdrawalLiveness, "requires withdrawalLiveness");
+
+  // Function which packs the arguments for a liquidation.
+  // returns parameters for empContract.methods.createLiquidation
+  function createLiquidationParams({ sponsor, tokensToLiquidate, currentBlockTime, maxCollateralPerToken }) {
+    assert(sponsor, "requires sponsor to liquidate");
+    assert(tokensToLiquidate, "requires tokenToLiquidate");
+    assert(currentBlockTime >= 0, "requires currentBlockTime");
+    assert(maxCollateralPerToken, "requires maxCollateralPerToken");
+    return [
+      sponsor,
+      { rawValue: liquidationMinPrice.toString() },
+      { rawValue: maxCollateralPerToken.toString() },
+      { rawValue: tokensToLiquidate.toString() },
+      parseInt(currentBlockTime) + liquidationDeadline
+    ];
+  }
+
+  function min(a, b) {
+    return toBN(a).lte(toBN(b)) ? a : b;
+  }
+  function max(a, b) {
+    return toBN(a).gte(toBN(b)) ? a : b;
+  }
```

Comment:
nit: I think these are already baked into the BN class as static methods. You can use:
```
const { BN } = web3.utils;
BN.min(x,y);
BN.max(x,y);
```

or to exactly replicate what you have here:
```
const { min, max } = web3.utils.BN;
```

Reply:
Oh wait, sorry @daywiss, I didn't realize you called `toBN` on these when they get passed in. That means the BN min and max may be a little different. You custom functions could make sense even if they just do the conversion and then pass to the BN methods.

Reply:
id rather use the BN versions. Did not realize it was there